File: test-project/greeter.javaexec/src/main/java/demo/Demo2.java
package demo;

import examples.greeter.api.Greeter;
import java.util.ServiceLoader;

/**
 * Test docs
 */
public class Demo2 {
    /**
     * Test docs
     * @param args test docs
     */
    public static void main(String[] args) {
        Greeter greeter = ServiceLoader.load(Greeter.class).findFirst().orElseThrow(() -> new RuntimeException("No Greeter found!"));
        System.out.println("Demo2: " + greeter.hello());
    }
}

File: test-project-kotlin/greeter.javaexec/src/main/kotlin/demo/Demo2.kt
package demo

import examples.greeter.api.Greeter

import java.util.ServiceLoader

fun main(args: Array<String>) {
    val greeter = ServiceLoader.load(Greeter::class.java)
            .findFirst()
            .orElseThrow { RuntimeException("No Greeter found!") }
    println("Demo2: " + greeter.hello())
}
